### DISCUSSION TITLE
SQLite connection flags

### DISCUSSION THREAD

**Discussion by Panda-Sharp** [linq2db/linq2db#3395] [Q&A] [answered]
Hello,

I tried to search everywhere but I couldn't find anything, is it possible to add flags to a SQLIte connection? if yes how?

For example with sqlite-net you can do 
```cs
using (var dbConnection = new DbConnection(SQLiteOpenFlags.FullMutex | SQLiteOpenFlags.ReadWrite;))
```

to be more precise, I'm trying to set this flag `SQLITE_THREADSAFE`
https://sqlite.org/threadsafe.html

thanks!

**Comment by MaceWindu** [comment, marked as answer]
Well, I don't think any of supported by linq2db providers (system.data.sqlite and microsoft.data.sqlite) support SQLITE_THREADSAFE option configuration and should use serialized mode by default.

**Reply by MaceWindu** [reply]
But if you find a way to configure connection somehow, then you can use it in linq2db with any `DataConnection` constructor that accepts [connection](https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/Data/DataConnection.cs#L156) or [connection factory](https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/Data/DataConnection.cs#L130).

**Comment by MaceWindu** [comment]
I've checked `sqlite-net` provider also - there is no way we will ever support it in linq2db as it is not an ADO.NET provider, which is strange

**Comment by Panda-Sharp** [comment]
Thanks @MaceWindu I'll take a look at your feedbacks !
